aiida_pseudo/cli/params/types.py

```diff
@@ -5,7 +5,45 @@
 
 from aiida.cmdline.params.types import GroupParamType
 
-__all__ = ('PseudoPotentialFamilyTypeParam', 'PseudoPotentialFamilyParam')
+__all__ = ('PseudoPotentialFamilyTypeParam', 'PseudoPotentialFamilyParam', 'PseudoPotentialTypeParam')
+
+
+class PseudoPotentialTypeParam(click.ParamType):
+    """Parameter type for `click` commands to define a subclass of `PseudoPotentialData`."""
+
+    name = 'pseudos_type_type'
+
+    def convert(self, value, _, __):
+        """Convert the entry point name to the corresponding class.
+
+        :param value: entry point name that should correspond to subclass of `PseudoPotentialFamily` group plugin
+        :return: the `PseudoPotentialFamily` subclass
+        :raises: `click.BadParameter` if the entry point cannot be loaded or is not subclass of `PseudoPotentialFamily`
+        """
+        from aiida.common import exceptions
+        from aiida.plugins import DataFactory
+        from aiida_pseudo.data.pseudo import PseudoPotentialData
+
+        try:
+            pseudo_type = DataFactory(value)
+        except exceptions.EntryPointError as exception:
+            raise click.BadParameter(f'`{value}` is not an existing data plugin.') from exception
+
+        if not issubclass(pseudo_type, PseudoPotentialData):
+            raise click.BadParameter(f'`{value}` entry point is not a subclass of `PseudoPotentialData`.')
+
+        PseudoPotentialData.entry_point = value
```

Comment:
I noticed that this line is also there for `PseudoPotentialFamilyParam` but I don't think it does anything and so is better to be removed

Reply:
This would make fail the tests `tests/cli/test_list.py::test_list_filter`and `tests/cli/test_list.py::test_list_filter_no_result`

Reply:
Ah I remember now, I added the original value that the user specifies on the param, because the converted value will be the class that correspond to the plugin. However, in the actual `list` command, I need the original entry point. Ok, good to keep then! :smile: